underfs/cos/src/main/java/alluxio/underfs/cos/COSUnderFileSystem.java

```diff
@@ -0,0 +1,310 @@
+
+  /** Aliyun COS client. */
+  private final COSClient mClient;
+
+  /** Bucket name of user's configured Alluxio bucket. */
+  private final String mBucketName;
+
+  /** Bucket name of user's configured Alluxio bucket. */
+  private final String mBucketNameInternal;
+  /**
+   * Constructs a new instance of {@link COSUnderFileSystem}.
+   *
+   * @param uri the {@link AlluxioURI} for this UFS
+   * @param conf the configuration for this UFS
+   * @return the created {@link COSUnderFileSystem} instance
+   */
+  public static COSUnderFileSystem createInstance(AlluxioURI uri,
+      UnderFileSystemConfiguration conf) throws Exception {
+    String bucketName = UnderFileSystemUtils.getBucketName(uri);
+    Preconditions.checkArgument(
+        conf.containsKey(PropertyKey.COS_ACCESS_KEY),
+        "Property " + PropertyKey.COS_ACCESS_KEY + " is required to connect to COS");
+    Preconditions.checkArgument(
+        conf.containsKey(PropertyKey.COS_SECRET_KEY),
+        "Property " + PropertyKey.COS_SECRET_KEY + " is required to connect to COS");
+    Preconditions.checkArgument(
+        conf.containsKey(PropertyKey.COS_REGION_KEY),
+        "Property " + PropertyKey.COS_REGION_KEY + " is required to connect to COS");
+    Preconditions.checkArgument(
+            conf.containsKey(PropertyKey.COS_APPID_KEY),
+            "Property " + PropertyKey.COS_APPID_KEY + " is required to connect to COS");
+    String accessId = conf.getValue(PropertyKey.COS_ACCESS_KEY);
+    String accessKey = conf.getValue(PropertyKey.COS_SECRET_KEY);
+    String regionName = conf.getValue(PropertyKey.COS_REGION_KEY);
+    String appId = conf.getValue(PropertyKey.COS_APPID_KEY);
+
+    COSCredentials cred = new BasicCOSCredentials(accessId, accessKey);
+    ClientConfig clientConfig = initializeCOSClientConfig(regionName);
+    COSClient client = new COSClient(cred, clientConfig);
+
+    return new COSUnderFileSystem(uri, client, bucketName, appId, conf);
+  }
+
+  /**
+   * Constructor for {@link COSUnderFileSystem}.
+   *
+   * @param uri the {@link AlluxioURI} for this UFS
+   * @param client Aliyun COS client
+   * @param bucketName bucket name of user's configured Alluxio bucket
+   * @param conf configuration for this UFS
+   */
+  protected COSUnderFileSystem(AlluxioURI uri, COSClient client, String bucketName, String appId,
+      UnderFileSystemConfiguration conf) {
+    super(uri, conf);
+    mClient = client;
+    mBucketName = bucketName;
+    mBucketNameInternal = bucketName + "-" + appId;
+  }
+
+  @Override
+  public String getUnderFSType() {
+    return "ocs";
+  }
+
+  // No ACL integration currently, no-op
+  @Override
+  public void setOwner(String path, String user, String group) {}
+
+  // No ACL integration currently, no-op
+  @Override
+  public void setMode(String path, short mode) throws IOException {}
+
+  @Override
+  protected boolean copyObject(String src, String dst) {
+    try {
+      LOG.info("Copying {} to {}", src, dst);
```

Comment:
`LOG.debug` please to avoid overhead on operations.

Reply:
Got it.